From bf4d5d86af5eff7c13a3ec0aac365ba7be141618 Mon Sep 17 00:00:00 2001
From: Peter Lemenkov <lemenkov@gmail.com>
Date: Thu, 3 Jul 2014 15:45:30 +0400
Subject: [PATCH 12/12] Allow passing directories as the config sources

We should allow passing the entire directories as the config sources. In
this case CouchDB will traverse the given directories and load every
*.ini file found.

E.g.

"... -couch_ini /etc/couchdb/default.d/ /etc/couchdb/local.d/ /path/to/extra/couchdb/default.ini /path/to/extra/couchdb/local.ini ..."

Signed-off-by: Peter Lemenkov <lemenkov@gmail.com>
---
 src/couchdb/couch_app.erl | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/src/couchdb/couch_app.erl b/src/couchdb/couch_app.erl
index 42411a8..7a4cdc3 100644
--- a/src/couchdb/couch_app.erl
+++ b/src/couchdb/couch_app.erl
@@ -15,6 +15,7 @@
 -behaviour(application).
 
 -include("couch_db.hrl").
+-include_lib("kernel/include/file.hrl").
 
 -export([start/2, stop/1]).
 
@@ -37,7 +38,17 @@ get_ini_files(Default) ->
     {ok, [[]]} ->
         Default;
     {ok, [Values]} ->
+        lists:foldl(
+        fun (V, AccIn) ->
+              case file:read_file_info(V) of
+                  {ok, #file_info{type = regular}} -> AccIn ++ [V];
+                  {ok, #file_info{type = directory}} -> AccIn ++ filelib:wildcard(V ++ "/*.ini");
+                  _ -> AccIn
+              end
+        end,
+        [],
         Values
+    )
     end.
 
 start_apps([]) ->
-- 
1.9.3

